agent/consul/state/catalog_test.go

```diff
@@ -3636,6 +3635,449 @@ func TestStateStore_CheckConnectServiceNodes_Gateways(t *testing.T) {
 	assert.Equal(443, nodes[1].Service.Port)
 }
 
+func TestStateStore_CheckGatewayServiceNodes_Terminating(t *testing.T) {
+	assert := assert.New(t)
+	s := testStateStore(t)
+
+	// Listing with no results returns an empty list.
+	ws := memdb.NewWatchSet()
+	idx, nodes, err := s.CheckGatewayServiceNodes(ws, "gateway", nil)
+	assert.Nil(err)
```

Comment:
Just a suggestion but if these were `assert.NoError` all throughout this test it might make it a little clearer what the intention is. Using `NoError` essentially performs the same check but with easier readability and better output when the assertion fails.